packages/cactus-cmd-api-server/src/test/typescript/integration/runtime-plugin-imports.test.ts

```diff
@@ -35,9 +49,92 @@ test("can import plugins at runtime (CLI)", async (t: Test) => {
     config: config.getProperties(),
   });
 
+  const startResponse = apiServer.start();
   await t.doesNotReject(
-    apiServer.start(),
+    startResponse,
     "failed to start API server with dynamic plugin imports configured for it...",
   );
+  t.ok(startResponse, "startResponse truthy OK");
+
+  const addressInfoApi = (await startResponse).addressInfoApi;
+  const protocol = apiServerOptions.apiTlsEnabled ? "https" : "http";
+  const { address, port } = addressInfoApi;
+  const apiHost = `${protocol}://${address}:${port}`;
+  t.comment(
+    `Metrics URL: ${apiHost}/api/v1/api-server/get-prometheus-exporter-metrics`,
+  );
+
+  const apiClient = new ApiServerApi({ basePath: apiHost });
+  apiServer.prometheusExporter.setTotalPluginImports(
+    apiServer.getPluginImportsCount(),
+  );
+
+  {
+    const res = await apiClient.getPrometheusExporterMetricsV1();
+    const promMetricsOutput =
+      "# HELP " +
+      K_CACTUS_API_SERVER_TOTAL_PLUGIN_IMPORTS +
+      " Total number of plugins imported\n" +
+      "# TYPE " +
+      K_CACTUS_API_SERVER_TOTAL_PLUGIN_IMPORTS +
+      " gauge\n" +
+      K_CACTUS_API_SERVER_TOTAL_PLUGIN_IMPORTS +
+      '{type="' +
+      K_CACTUS_API_SERVER_TOTAL_PLUGIN_IMPORTS +
+      '"} 1';
+    t.ok(res);
+    t.ok(res.data);
+    t.equal(res.status, 200);
+    t.true(
+      res.data.includes(promMetricsOutput),
+      "Total 1 plugins imported as expected. RESULT OK",
+    );
+  }
+
+  // Adding a new plugin to update the prometheus metric K_CACTUS_API_SERVER_TOTAL_PLUGIN_IMPORTS
+  const keyPair = await JWK.generate("EC", "secp256k1", { use: "sig" }, true);
+  const keyPairPem = keyPair.toPEM(true);
+  const db: ConsortiumDatabase = {
+    cactusNode: [],
+    consortium: [],
+    consortiumMember: [],
+    ledger: [],
+    pluginInstance: [],
+  };
+
+  const options: IPluginConsortiumManualOptions = {
+    instanceId: uuidv4(),
+    keyPairPem: keyPairPem,
+    consortiumDatabase: db,
+  };
+
+  const pluginConsortiumManual: PluginConsortiumManual = new PluginConsortiumManual(
+    options,
+  );
+
+  (await apiServer.getOrInitPluginRegistry()).add(pluginConsortiumManual);
```

Comment:
@jagpreetsinghsasan 

I recommend separating the two invocations into their own line so that if they throw in production and all we have is a line number in a log then it is clear which method threw.

```typescript
const pluginRegistry = await apiServer.getOrInitPluginRegistry();
pluginRegistry.add(pluginConsortiumManual);
```